Updating guide’s content to use the Liberty AKS portal for creating cluster, resource group, and container

Ruilin-Ma opened the pull request:
issues: OpenLiberty/guides-common#1107 and https://github.com/OpenLiberty/guides-common/issues/1068

Ruilin-Ma committed:
updating guide’s content to use the liberty AKS portal for creating cluster, resource group, and container

Ruilin-Ma committed:
Merge branch 'liberty-aks-update' of https://github.com/OpenLiberty/guide-cloud-azure into liberty-aks-update

Ruilin-Ma committed:
lint correction & minor content fix

Ruilin-Ma committed:
Resolve image display issue & content improvement

Ruilin-Ma committed:
Image update & Restruct the content layout

Ruilin-Ma committed:
Revise the subtitle

Ruilin-Ma committed:
Guide's content improvement & Image improvement

Ruilin-Ma committed:
Fixed typo

gkwan-ibm commented on README.adoc:
```diff
@@ -56,13 +57,9 @@ A cloud-based infrastructure enables you to focus on developing your microservic
-https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/[AKS pricing^] documentation for more details.
+Azure offers a managed {kube} service called **Azure Kubernetes Service** (AKS). Using AKS simplifies the process of running Kubernetes on Azure without needing to install or maintain your own Kubernetes control plane. It provides a hosted {kube} cluster that you can deploy your microservices to. You will use AKS with an **Azure Container Registry** (ACR). ACR is a private registry that is used t ...
+
+This guide will introduce **IBM WebSphere Liberty and Open Liberty on AKS**, which solution automates the majority of standard steps involved in provisioning WebSphere Liberty or Open Liberty on an AKS cluster. Once the initial provisioning process is complete, you are free to customize deployments further.
```
**IBM WebSphere Liberty and Open Liberty on AKS** -> **IBM WebSphere Liberty and Open Liberty on AKS** plan

gkwan-ibm commented on README.adoc:
```diff
@@ -87,11 +84,15 @@ If you already have Docker installed, make sure to have it running.
 https://azure.microsoft.com/en-us/pricing/purchase-options/pay-as-you-go/[Microsoft Azure Purchase Options^] 
 to create an account with your email and start a Pay-As-You-Go subscription. 
 
+* *IBM WebSphere Liberty and Open Liberty on AKS:* You will need to use the IBM WebSphere Liberty and Open Liberty on AKS on Azure marketplace. See the official
```
"...the IBM WebSphere Liberty and Open Liberty on AKS on Azure marketplace..." -> "...the IBM WebSphere Liberty and Open Liberty on AKS plan on Azure marketplace..."

gkwan-ibm commented on README.adoc:
```diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
-To deploy your microservices, you need to create an Azure Container Registry in the same location 
-where your services are deployed, and link the registry to a resource group. Your registry will manage container instances 
-that will be deployed to a Kubernetes cluster.
+== Provision deployment foundation with IBM Liberty on AKS 
```
"IBM Liberty" -> "Liberty"

gkwan-ibm commented on README.adoc:
```diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
 
-A resource group is an Azure construct to manage a logical collection of resources for your cloud deployments on Azure. 
-You must create a new resource group to manage the resources you need for your Kubernetes deployment.
+To create the resource group in an efficient way, check out the https://aka.ms/liberty-aks[IBM Liberty on AKS] tool and click on the `Create` button to start the process. 
```
"...[IBM Liberty on AKS] tool ..." -> "...[IBM WebSphere Liberty and Open Liberty on AKS] plan..."

gkwan-ibm commented on README.adoc:
```diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
-To create a resource group, an Azure location must be specified. The metadata for your resources are stored at this specified Azure location. 
-If resources are created later without specifying a location, these new resources run in the
-same region that you specified for creating a resource group. 
+On the `Basics` pane. Under the `Resource group`, select `Create new` and type in **guideGroup** as the name of the resource group. Please note, the offer also supports to use an existing resource group while the group is required to be empty. 
```
- "...select `Create new`..." -> "...click on the `Create new` link ..."
- **guideGroup** -> `guideGroup`

gkwan-ibm commented on README.adoc:
````diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
-```
-az account list-locations -o table
-```
+An Azure location must be specified. The metadata for your resources are stored at this specified Azure location. If resources are created later without specifying a location, these new resources will run in the same region that you specified for creating a resource group. Click on the `Region` drop down menu and select the desired location.
````
- insert a screenshot for the Region selection

gkwan-ibm commented on README.adoc:
```diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
-You will see an output similar to the following:
+However, AKS is not available in all regions. Make sure that the region you select is 
+https://azure.microsoft.com/en-us/global-infrastructure/services/?products=kubernetes-service[compatible with AKS^].
 
```
insert a screenshot for the Region selection

gkwan-ibm commented on README.adoc:
```diff
@@ -123,130 +124,52 @@ include::{common-includes}/gitclone.adoc[]
+Upon completion of the deployment process, navigate to the resource group overview page in the Azure Portal. Review the created resource group and search for resources of the type `Container registry`. Locate the container registry that was created during the deployment. The name of the Container Registry will be used in `[registry-name]`.
 
-You will see an output similar to the following:
+image::Locate_ACR.png[Locate ACR,align="center"]
```
- can make the text bigger in this screenshot?

Ruilin-Ma committed:
Comments addressed

gkwan-ibm committed:
Dependabot Bump (#134)

* Bump org.apache.maven.plugins:maven-failsafe-plugin in /start

Bumps [org.apache.maven.plugins:maven-failsafe-plugin](https://github.com/apache/maven-surefire) from 3.3.0 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.0...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-failsafe-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump org.apache.maven.plugins:maven-surefire-plugin in /start

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.3.0 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.0...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump org.apache.maven.plugins:maven-failsafe-plugin in /finish

Bumps [org.apache.maven.plugins:maven-failsafe-plugin](https://github.com/apache/maven-surefire) from 3.3.0 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.0...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-failsafe-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump org.apache.maven.plugins:maven-surefire-plugin in /finish

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.3.0 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.0...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gkwan-ibm commented:
- search the role requirement from azure doc for users to create resource group, then add those prereq in the content

Ruilin-Ma committed:
Improving the image quality and content based on the comments

Ruilin-Ma committed:
Merge branch 'liberty-aks-update' of https://github.com/OpenLiberty/guide-cloud-azure into liberty-aks-update

gkwan-ibm commented:
- [x] It is better to simplify the content of the **Provision deployment foundation with Liberty on AKS** section. The guide has no need to describe in detail like as a doc. Users only need to provide info in the Basics pane and leave other panes as default. After filling up the Basics pane, they can click the **Review+Create** button to create the resources group and cluster. Let the guide to demonstrate how simple to create them. Just mention that if interest to know more configuration, they can visit the other panes.
- [x] in the **Access the created cluster** section, is there a command, e.g. `az aks show`, to get the `[cluster-name]` instead of from the GUI?
- [x] `clusterXXXXXX` in the output can be replaced by a makeup string, e.g. `cluster1234d5`
- [x] in the **Logging into the container registry** subsection, can add a command to get the `[registry-name]`? In this case, screenshot can be kept.

Ruilin-Ma committed:
Simplify the content of the Deployment section & replace cluster name retrieval with command line instead of GUI

Ruilin-Ma committed:
Content improvement

Ruilin-Ma committed:
example output update

Ruilin-Ma committed:
content improvement

Ruilin-Ma committed:
content improvement

Ruilin-Ma committed:
adding a command line to obtain [registry-name] && addressing comments

Ruilin-Ma committed:
content improvement

Ruilin-Ma committed:
revert unwanted changes

gkwan-ibm commented:
- [x] remove
  - > Please note, the plan also supports to use an existing resource group while the group is required to be empty. 
- [x] Before this statement "To push Docker images to your registry, you must log in to your Azure Container Registry by using the Azure CLI.", need to tell users which value is the `[registry-name]`, e.g. acr1234d5 under the `NAME` column from the output

Ruilin-Ma committed:
Addressing comment & improving the instruction for replacing [registry-name]

gkwan-ibm committed:
Update README.adoc

gkwan-ibm committed:
Update README.adoc

Ruilin-Ma committed:
Merge pull request #135 from OpenLiberty/gkwan-dev-content-review

Dev content review

gkwan-ibm committed:
Merge branch 'staging' into liberty-aks-update

shin19991207 commented:
Address Laura's comments:
- [x] Can you remove the bold highlighting on the names in the What you’ll learn and the Additional prereqs sections, pls? It’s not necessary.
- [x] “a cost may associated” > “a cost may be associated”
- [x] “Azure Subscription:” should be “Azure subscription:“.
- [x] “IBM WebSphere Liberty and Open Liberty on AKS:” should be “IBM WebSphere Liberty and Open Liberty on AKS plan:” But also remove all the bold highlighting here.
- [x] “Provision deployment foundation with Liberty on AKS” and others sections need to have some kind of introduction for the section before the first sub-heading (see old Azure guide for example, though it too doesn’t always do it but should).
- [x] Not sure about the heading “Provision deployment foundation with Liberty on AKS” - could do with something more meaningful I think.

shin19991207 committed:
Addressed Laura's comments and update the guide to use OL operator to deploy the app

shin19991207 committed:
Update deploy.yaml

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update deploy.yaml

shin19991207 committed:
Update README.adoc

gkwan-ibm committed:
Merge branch 'staging' into liberty-aks-update

shin19991207 committed:
Update test.yml

Set VALIDATE_KUBERNETES_KUBEVAL to false

gkwan-ibm commented:
- [x] Somewhere need to mention the plan/provisioning of the cluster also provides the OL operator.
- [x] test on Win
- [x] on Mac, the `az aks install-cli` command need to use `sudo az aks install-cli`. Linix may need `sudo` too. Surely it may different than Win.
- [x] at the end of the **Setting up ACR and Kubernetes cluster on AKS** subsection, tell users how to know the deployment is completed, e.g. When the deployment is completed, a `Your deployment is complete` message is shown.
- [x] at the end of the **Connecting the AKS cluster to Kubernetes** subsection, update the output `v1.29.7` to `v1.29.8`

shin19991207 commented:
> - [x] on Mac, the `az aks install-cli` command need to use `sudo az aks install-cli`. Linix may need `sudo` too. Surely it may different than Win.

- [Sudo for Windows](https://learn.microsoft.com/en-us/windows/sudo/) requires Windows 11 Insider Preview Build 26052 or higher
- [Runas](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-r2-and-2012/cc771525(v=ws.11))
  - The `runas` command in Windows requires running with a full path, i.e. `runas /user:Administrator "C:\path\to\az.cmd aks install-cli"`
  - Every Windows installation has a built-in Administrator account. However, it is not always enabled by default.
  - Alternatively, instead of using `runas`, you can open Command Prompt or PowerShell as Administrator (right-click > "Run as administrator") and then run the command(s)

shin19991207 committed:
Addressed dev content review

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Fixed superlinter error

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

shin19991207 committed:
Update README.adoc

gkwan-ibm committed:
Update README.adoc

gkwan-ibm committed:
Merge branch 'staging' into liberty-aks-update

gkwan-ibm committed:
Merge branch 'staging' into liberty-aks-update

gkwan-ibm committed:
Merge branch 'staging' into liberty-aks-update

gkwan-ibm committed:
Update README.adoc

gkwan-ibm committed:
Update README.adoc

dmuelle commented on README.adoc:
```diff
@@ -36,33 +38,22 @@ Explore how to deploy microservices to Azure Kubernetes Service (AKS) on Microso
-deploying, managing, and scaling containerized applications. If you would like to learn
-more about Kubernetes, check out the https://openliberty.io/guides/kubernetes-intro.html[Deploying microservices to Kubernetes^]
-guide.
+Kubernetes is an open source container orchestrator that automates many tasks involved in deploying, managing, and scaling containerized applications. If you would like to learn more about Kubernetes and how to use it with the Open Liberty Operator, check out the https://openliberty.io/guides/openliberty-operator-intro.html[Deploying a microservice to Kubernetes using Open Liberty Operator^] guid ...
```
```suggestion
Kubernetes is an open source container orchestrator that automates many tasks involved in deploying, managing, and scaling containerized applications. To learn more about Kubernetes and how to use it with the Open Liberty Operator, check out the https://openliberty.io/guides/openliberty-operator-intro.html[Deploying a microservice to Kubernetes using Open Liberty Operator^] guide.
```

dmuelle commented on README.adoc:
```diff
@@ -36,33 +38,22 @@ Explore how to deploy microservices to Azure Kubernetes Service (AKS) on Microso
-and distribute your container images. Note, because AKS is not free a small
-cost is associated with running this guide. See the official
-https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/[AKS pricing^] documentation for more details.
+Azure offers a managed {kube} service called Azure Kubernetes Service (AKS). Using AKS simplifies the process of running Kubernetes on Azure without needing to install or maintain your own Kubernetes control plane. It provides a hosted {kube} cluster that you can deploy your microservices to. You will use AKS with an Azure Container Registry (ACR). ACR is a private registry that is used to store  ...
```
```suggestion
Azure offers a managed {kube} service called Azure Kubernetes Service (AKS). Using AKS simplifies the process of running Kubernetes on Azure without needing to install or maintain your own Kubernetes control plane. It provides a hosted {kube} cluster that you can deploy your microservices to. You will use AKS with Azure Container Registry (ACR). ACR is a private registry that is used to store and distribute your container images. Because AKS is not free, a cost might be associated with running this guide. See the official https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/[AKS pricing^] documentation for more details.
```

dmuelle commented on README.adoc:
```diff
@@ -36,33 +38,22 @@ Explore how to deploy microservices to Azure Kubernetes Service (AKS) on Microso
-https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/[AKS pricing^] documentation for more details.
+Azure offers a managed {kube} service called Azure Kubernetes Service (AKS). Using AKS simplifies the process of running Kubernetes on Azure without needing to install or maintain your own Kubernetes control plane. It provides a hosted {kube} cluster that you can deploy your microservices to. You will use AKS with an Azure Container Registry (ACR). ACR is a private registry that is used to store  ...
+
+This guide will introduce IBM WebSphere Liberty and Open Liberty on AKS plan, a solution that automates most of the standard steps for provisioning WebSphere Liberty or Open Liberty on an AKS cluster. Once the initial provisioning is complete, you can further customize your deployments as needed.
```
```suggestion
This guide will introduce the IBM WebSphere Liberty and Open Liberty on AKS plan, a solution that automates most of the standard steps for provisioning WebSphere Liberty or Open Liberty on an AKS cluster. Once the initial provisioning is complete, you can further customize your deployments as needed.
```

dmuelle commented on README.adoc:
```diff
@@ -83,15 +74,19 @@ supports various container types, but you will use Docker in this guide. For ins
 If you already have Docker installed, make sure to have it running.
 
-* *Azure Subscription:* To run this guide, you will need an Azure subscription. Navigate to the 
+* *Azure subscription:* To run this guide, you will need an Azure subscription. Navigate to the 
```
```suggestion
* *Azure subscription*: To run this guide, you will need an Azure subscription. Navigate to the 
```

dmuelle commented on README.adoc:
```diff
@@ -83,15 +74,19 @@ supports various container types, but you will use Docker in this guide. For ins
 https://azure.microsoft.com/en-us/pricing/purchase-options/pay-as-you-go/[Microsoft Azure Purchase Options^] 
 to create an account with your email and start a Pay-As-You-Go subscription. 
 
+* *IBM WebSphere Liberty and Open Liberty on AKS plan:* You will need to use the IBM WebSphere Liberty and Open Liberty on AKS plan on Azure marketplace. See the official
```
```suggestion
* *IBM WebSphere Liberty and Open Liberty on AKS plan*: You will need to use the IBM WebSphere Liberty and Open Liberty on AKS plan on Azure marketplace. See the official
```

dmuelle commented on README.adoc:
```diff
@@ -101,11 +96,28 @@ az --version
 * *kubectl:* You need the Kubernetes command-line tool `kubectl` to interact with your Kubernetes cluster.
```
```suggestion
* *kubectl*: You need the Kubernetes command-line tool `kubectl` to interact with your Kubernetes cluster.
```

dmuelle commented on README.adoc:
```diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
 
 === Creating a resource group
 
+A resource group is an Azure construct to manage a logical collection of resources for your cloud deployments on Azure. You must create a new resource group to hold and manage the resources you need for your Kubernetes deployment.
```
```suggestion
A resource group is an Azure construct to manage a logical collection of resources for your cloud deployments on Azure. You must create a resource group to hold and manage the resources you need for your Kubernetes deployment.
```

dmuelle commented on README.adoc:
```diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
 
-However, AKS is not available in all regions. Make sure that the region you select is 
-https://azure.microsoft.com/en-us/global-infrastructure/services/?products=kubernetes-service[compatible with AKS^].
+After setting up the resource group, you can leave the remaining configuration at its default settings by skipping the **AKS**, **Load balancing**, and **Operator and application** panes. The default configuration will automatically set up an https://azure.microsoft.com/en-ca/products/container-registry[Azure Container Registry^] in the same region where your services are deployed, link it to you ...
```
```suggestion
After you set up the resource group, leave the remaining configuration at its default settings by skipping the **AKS**, **Load balancing**, and **Operator and application** panes. The default configuration automatically sets up an https://azure.microsoft.com/en-ca/products/container-registry[Azure Container Registry^] in the same region where your services are deployed, links it to your resource group, creatse an https://learn.microsoft.com/en-us/azure/aks/tutorial-kubernetes-deploy-cluster?tabs=azure-cli[AKS cluster^], and deploys the https://github.com/OpenLiberty/open-liberty-operator/releases/latest[Open Liberty Operator^] to manage your Open Liberty applications.
```

dmuelle commented on README.adoc:
```diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
 
-Create a resource group using the `az group create` command. Remember to replace `[location]` with a region 
-that is available for your subscription and compatible with AKS.
+For more information on configuring these options, refer to the https://learn.microsoft.com/en-us/azure/aks/howto-deploy-java-liberty-app?tabs=in-bash#create-a-liberty-on-aks-deployment-using-the-portal[Create a Liberty on AKS deployment using the portal] documentation.
```
```suggestion
For more information on configuring these options, see the https://learn.microsoft.com/en-us/azure/aks/howto-deploy-java-liberty-app?tabs=in-bash#create-a-liberty-on-aks-deployment-using-the-portal[Create a Liberty on AKS deployment using the portal] documentation.
```

dmuelle commented on README.adoc:
````diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
-```
-az group create -l [location] -n guideGroup
-```
+Once you have reviewed the configurations, select **Review + create** to validate your setup. After validation, click **Create** to initiate the deployment, which can take up to 20 minutes to complete. You'll see a `Deployment is in progress` message, and when the deployment is complete, a `Your deployment is complete` message will confirm the creation of the ACR and AKS cluster.
````
```suggestion
After you review the configurations, select **Review + create** to validate your setup. After validation, click **Create** to initiate the deployment, which can take up to 20 minutes to complete. You'll see a `Deployment is in progress` message, and when the deployment is complete, a `Your deployment is complete` message will confirm the creation of the ACR and AKS cluster.
```

dmuelle commented on README.adoc:
````diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
-  "type": null
-}
-```
+Once your Azure Container Registry (ACR) is created, you can proceed to containerize your microservices and upload their Docker images to the ACR for storage and deployment.
````
```suggestion
Once your Azure Container Registry (ACR) is created, you can containerize your microservices and upload their Docker images to the ACR for storage and deployment.
```

dmuelle commented on README.adoc:
````diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
-```
-az acr create -g guideGroup -n [registry-name] --sku Basic --admin-enabled
-```
+For example, in the figure below, the registry name is `acrc75e71`.
````
```suggestion
For example, in the following screenshot, the registry name is `acrc75e71`.
```

dmuelle commented on README.adoc:
```diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
-The possible Stock Keeping Unit (SKU) values that can be passed into the `--sku` option are `Basic`, `Standard`, and `Premium`.
-These different SKU options provide pricing for various levels of capacity and usage. You use a Basic SKU because it is cheaper 
-and the services you deploy have low storage and throughput requirements.
+You will use this registry name in place of `[registry-name]` in the upcoming commands.
```
```suggestion
Use this registry name in place of `[registry-name]` in the upcoming commands.
```

dmuelle commented on README.adoc:
```diff
@@ -123,154 +135,78 @@ include::{common-includes}/gitclone.adoc[]
-which is `[registry-name].azurecr.io`, with all lowercase letters.
 
-=== Logging into the container registry
+This command will return a table with the details of your ACR, including its name. For example, you will see an output similar to the following:
```
```suggestion
This command returns a table with the details of your ACR, including its name. Look for output similar to the following example:
```

dmuelle commented on README.adoc:
````diff
@@ -307,7 +239,7 @@ To verify that the images are built, run the `docker images` command to list all
 ```
 
-Your two images `system` and `inventory` should appear in the list of all Docker images:
+Your two images, `system` and `inventory`, should appear in the list of all Docker images:
````
```suggestion
Your two images, `system` and `inventory`, are included in the list of all Docker images:
```

dmuelle commented on README.adoc:
```diff
@@ -320,18 +252,29 @@ system        1.0-SNAPSHOT    1dff6d0b4f31    5 minutes ago    470MB
-Replace `[registry-server]` with the server name of your container registry.
-To get the server for your registry, run the `az acr show -n [registry-name] --query loginServer` command.
-The `[registry-server]` looks like `[registry-name].azurecr.io`.
+First, retrieve your registry server name. This will typically be in the format `[registry-name].azurecr.io`. You can find this by running the following command:
```
```suggestion
First, retrieve your registry server name, which typically be in the following format: `[registry-name].azurecr.io`. You can find the server name by running the following command:
```

dmuelle commented on README.adoc:
````diff
@@ -320,18 +252,29 @@ system        1.0-SNAPSHOT    1dff6d0b4f31    5 minutes ago    470MB
+az acr show -n [registry-name] --query loginServer
+```
+
+You will see an output similar to the following that will be used to substitute the `[registry-server]` value of the commands in below.
````
```suggestion
Look for output similar to the following example. 
```

dmuelle commented on README.adoc:
```diff
@@ -320,18 +252,29 @@ system        1.0-SNAPSHOT    1dff6d0b4f31    5 minutes ago    470MB
+"acr31a3a1.azurecr.io"
+----
+
+Tag your container images with your registry by the following commands:
```
```suggestion
Tag your container images with your registry by the following commands. Substitute your registry server name for the `[registry-server]` value.
```

dmuelle commented on README.adoc:
````diff
@@ -320,18 +252,29 @@ system        1.0-SNAPSHOT    1dff6d0b4f31    5 minutes ago    470MB
 ```
 
-Finally, push your images to the registry:
+Finally, push your images to the registry by the following commands:
````
```suggestion
Finally, push your images to the registry by running the following commands:
```

dmuelle commented on README.adoc:
```diff
@@ -340,48 +283,44 @@ docker push [registry-server]/inventory:1.0-SNAPSHOT
+=== Connecting the AKS cluster to Kubernetes
 
-=== Provisioning a cluster
+To manage and interact with your AKS cluster using Kubernetes, you need to integrate the AKS cluster credentials into your local Kubernetes configuration. This allows your local Kubernetes CLI (`kubectl`) to communicate with the AKS cluster for deploying and managing resources.
```
```suggestion
To manage and interact with your AKS cluster using Kubernetes, you need to integrate the AKS cluster credentials into your local Kubernetes configuration. This integration allows your local Kubernetes CLI (`kubectl`) to communicate with the AKS cluster for deploying and managing resources.
```

dmuelle commented on README.adoc:
```diff
@@ -340,48 +283,44 @@ docker push [registry-server]/inventory:1.0-SNAPSHOT
-To create your AKS cluster, use the `az aks create` cluster command. 
-When the cluster is created, the command outputs information about the cluster.
-You might need to wait while your cluster is being created.
+To begin, obtain the information about the Kubernetes cluster created within the `guideGroup` resource group. You can do this by either viewing the resource group overview in the Azure Portal or executing the following command:
```
```suggestion
To begin, obtain the information about the Kubernetes cluster created within the `guideGroup` resource group. You can find this information by either viewing the resource group overview in the Azure Portal or running the following command:
```

dmuelle commented on README.adoc:
```diff
@@ -340,48 +283,44 @@ docker push [registry-server]/inventory:1.0-SNAPSHOT
 
-Running this command creates an AKS cluster that is called `guideCluster` with the resource group
-`guideGroup`.
+The cluster name, formatted as `clusterXXXXXX`, should be identified in the following output.
```
```suggestion
The cluster name, formatted as `clusterXXXXXX`, is identified in the following output.
```

dmuelle commented on README.adoc:
```diff
@@ -340,48 +283,44 @@ docker push [registry-server]/inventory:1.0-SNAPSHOT
 
-Merge the credentials of your cluster into your current Kubernetes configuration by using the `az aks get-credentials` command. 
-The default Kubernetes configuration file that is updated with your cluster credentials is located within the `~/.kube/config` filepath. 
+Merge the credentials of your cluster into your current Kubernetes configuration by using the `az aks get-credentials` command. Replacing `[cluster-name]` with the name of the cluster. The default Kubernetes configuration file that is updated with your cluster credentials is located within the `~/.kube/config` filepath.
```
```suggestion
Merge the credentials of your cluster into your current Kubernetes configuration by using the `az aks get-credentials` command. Replace `[cluster-name]` with the name of the cluster. The default Kubernetes configuration file that is updated with your cluster credentials is located within the `~/.kube/config` filepath.
```

dmuelle commented on README.adoc:
```diff
@@ -391,15 +330,12 @@ Run the following command to check the status of the available nodes in your AKS
 
-The `kubectl get nodes` command outputs information about three nodes, as the cluster was created with the default number of nodes in a node pool.
-The `STATUS` of each node is in the `Ready` state. 
+The `kubectl get nodes` command outputs information about the node. Please ensure that the `STATUS` of the node is in the `Ready` state before proceeding with the subsequent steps. 
```
```suggestion
The `kubectl get nodes` command outputs information about the node. Ensure that the `STATUS` of the node is in the `Ready` state before you proceed. 
```

dmuelle commented on README.adoc:
```diff
@@ -436,38 +395,31 @@ secret/guidesecret created
-in learning more about the Kubernetes resource definition, check out the 
-https://openliberty.io/guides/kubernetes-intro.html[Deploying microservices to Kubernetes^]
-guide.
+The Open Liberty Operator provides custom resource definitions (CRDs) that allow you to define how your Open Liberty applications are deployed. These CRDs are described in a `yaml` file, which outlines the deployment details and services required for your application. All resources can also be deleted from the cluster by using the same `yaml` file that you used to deploy them. The [hotspot]`deplo ...
```
```suggestion
The Open Liberty Operator provides custom resource definitions (CRDs) that define how your Open Liberty applications are deployed. These CRDs are described in a YAML file, which outlines the deployment details and services required for your application. You can also delete any resources from the cluster by using the same YAML file that you used to deploy them. In this guide, the [hotspot]`deploy.yaml` resource definition file is provided for you. To learn more about deploying microservices using the Open Liberty Operator, check out the https://openliberty.io/guides/openliberty-operator-intro.html[Deploying a microservice to Kubernetes using Open Liberty Operator^] guide.
```

dmuelle reviewed (changes requested):
ID review- let me now if you have any questions. Thanks

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

gkwan-ibm committed:
Update README.adoc

dmuelle reviewed (approved):
ID LGTM, thanks

gkwan-ibm committed:
Use versionless features (#164)

* use versionless feature

Signed-off-by: Gilbert Kwan <gkwan@ca.ibm.com>

* use versionless feature

Signed-off-by: Gilbert Kwan <gkwan@ca.ibm.com>

* use versionless feature

Signed-off-by: Gilbert Kwan <gkwan@ca.ibm.com>

---------

Signed-off-by: Gilbert Kwan <gkwan@ca.ibm.com>

gkwan-ibm reviewed (approved):
verified end-to-end and had no issue

yeekangc reviewed (approved):
Thanks, everyone!  Let us proceed to publish.
